modules/nf-core/maxbin2/main.nf

```diff
@@ -27,7 +28,15 @@ process MAXBIN2 {
     script:
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
-    def associate_files = reads ? "-reads $reads" : "-abund $abund"
+    def associate_files = ""
+    if ( reads ) {
+        associate_files = "-reads $reads"
+    } else if ( abund instanceof List ) {
+        associate_files = "-abund ${abund[0]}"
+        for (i in 2..abund.size()) { associate_files += " -abund$i ${abund[i-1]}" }
+    } else {
+        associate_files = "-abund $abund"
+    }
```

Comment:
Maybe also add an error check that `reads` and `abund` cannot be supplied together

> (Warning: Please do NOT specify abundance and reads that BELONG TO THE SAME METAGENOME together. MaxBin will treat them as two different information and thus will count this metagenome TWICE!)

From the docs

Reply:
But only ever reads or abund is used in the module, never both, so that isnt neccessary, if I am not mistaken?